gitops/snapshot.go

```diff
@@ -227,6 +241,18 @@ func IsSnapshotNotStarted(snapshot *applicationapiv1alpha1.Snapshot) bool {
 	return false
 }
 
+// IsSnapshotError if the AppStudio Integration Status condition is in ErrorOcurred status.
+func IsSnapshotError(snapshot *applicationapiv1alpha1.Snapshot) bool {
+	condition := meta.FindStatusCondition(snapshot.Status.Conditions, AppStudioIntegrationStatusCondition)
+	if condition == nil {
+		condition = meta.FindStatusCondition(snapshot.Status.Conditions, LegacyIntegrationStatusCondition)
+	}
+	if condition.Reason == AppStudioIntegrationStatusErrorOccured {
```

Comment:
Should we change it to `if condition == nil || condition.Reason == AppStudioIntegrationStatusErrorOccured` in case `LegacyIntegrationStatusCondition` can't be found either?

Reply:
If condition == nil, the expected return value is false.

Reply:
Ah, yes. You are right.